File: app_mvp/src/main/java/com/example/jqorz/test1/mvp/delegate/BaseDelegateCallback.java
package com.example.jqorz.test1.mvp.delegate;


import com.example.jqorz.test1.mvp.MvpView;
import com.example.jqorz.test1.mvp.presenters.MvpPresenter;

/**
 * Created by jqorz on 2017/8/28.
 * 用于对获取MvpView、创建以及获取Presenter
 */

public interface BaseDelegateCallback<P extends MvpPresenter, V extends MvpView> {

    void setPresenter();

    P getPresenter();

    P createPresenter();

    V getMvpView();

}

File: app_mvp/src/main/java/com/example/jqorz/mvptest/lib_kotlin/delegate/BaseDelegateCallback.kt
package com.jqorz.tallybook.base.mvp.delegate

import com.jqorz.tallybook.base.mvp.MvpView
import com.jqorz.tallybook.base.mvp.presenters.MvpPresenter

/**
 * 用于对获取MvpView、创建以及获取Presenter
 */
interface BaseDelegateCallback<P : MvpPresenter<V>?, V : MvpView> {
    fun setPresenter()
    fun getPresenter(): P?
    fun createPresenter(): P
    fun getMvpView(): V
}